onadata/apps/api/tools.py

```diff
@@ -340,10 +357,16 @@ def add_team_to_project(team, project):
 
 def publish_xlsform(request, owner, id_string=None, project=None):
     """
-    Publishes XLSForm given a request.
+    Publishes XLSForm & creates an XFormVersion object given a request.
     """
-    return do_publish_xlsform(request.user, request.data, request.FILES, owner,
-                              id_string, project)
+    survey = do_publish_xlsform(
+        request.user, request.data, request.FILES, owner, id_string, project)
+
+    # Create an XFormVersion object if the returned object is an
+    # XForm object
+    if isinstance(survey, XForm):
+        create_xform_version(survey, request.user)
```

Comment:
I wonder whether this should have been called or addressed at https://github.com/onaio/onadata/blob/2d039f2335bc3cf572f5923616f744e7360f4c6f/onadata/libs/utils/logger_tools.py#L655 and https://github.com/onaio/onadata/blob/2d039f2335bc3cf572f5923616f744e7360f4c6f/onadata/libs/utils/logger_tools.py#L683 such that all cases where a form is published will be captured. If we do publish an XForm with ODK Briefcase, will a version be created in this case?